src/protobuf/java/com/datadoghq/sketch/ddsketch/store/StoreProtoBinding.java

```diff
@@ -0,0 +1,60 @@
+package com.datadoghq.sketch.ddsketch.store;
+
+import java.util.function.Supplier;
+
+public class StoreProtoBinding {
+
+    /**
+     * Generates a protobuf representation of this {@code Store}.
+     *
+     * @return a protobuf representation of this {@code Store}
+     */
+    public static com.datadoghq.sketch.ddsketch.proto.Store toProto(Store store) {
+        if (store instanceof DenseStore) {
+            return toProtoDense((DenseStore) store);
+        }
+        return toProtoSparse(store);
+    }
+
+    private static com.datadoghq.sketch.ddsketch.proto.Store toProtoSparse(Store store) {
+        final com.datadoghq.sketch.ddsketch.proto.Store.Builder storeBuilder =
+                com.datadoghq.sketch.ddsketch.proto.Store.newBuilder();
+        // In the general case, we use the sparse representation to encode bin counts.
+        store.getStream().forEach(bin -> storeBuilder.putBinCounts(bin.getIndex(), bin.getCount()));
+        return storeBuilder.build();
+    }
+
+    private static com.datadoghq.sketch.ddsketch.proto.Store toProtoDense(DenseStore store) {
+        final com.datadoghq.sketch.ddsketch.proto.Store.Builder builder =
+                com.datadoghq.sketch.ddsketch.proto.Store.newBuilder();
+        if (store.counts != null) {
+            // For the dense store, we use the dense representation to encode the bin counts.
+            builder.setContiguousBinIndexOffset(store.minIndex);
+            for (long index = store.minIndex; index <= store.maxIndex; index++) {
```

Comment:
No need to change now, but I think I used `long` to handle the case `store.maxIndex == Integer.MAX_VALUE`, but I wonder about the cost now. Maybe should we favor what we do elsewhere, with `index < store.maxIndex` and the special handling of the last value.

Reply:
Good catch, I just copied and pasted this. Using `long` indexes in loops disables autovectorisation in hotspot.

Reply:
I changed this - I think it's fine to iterate `[store.minIndex - store.offset, store.maxIndex - store.offset]` because even if the upper bound is `Integer.MAX_VALUE`, that's 4 or 5 elements larger than any JVM will let you create.